tasks/move-granules/index.js

```diff
@@ -172,56 +176,140 @@ async function updateGranuleMetadata(granulesObject, collection, cmrFiles, bucke
   return updatedGranules;
 }
 
+/**
+ * Handles errors from the file lookup API call during cross-collection check.
+ *
+ * @param {unknown} error - The error thrown by the API call
+ * @param {string} bucket - S3 bucket name
+ * @param {string} key - S3 key
+ * @param {boolean} crossCollectionThrowOnNotFound - Whether to throw on 404 errors
+ * @throws {FileNotFound|Error} Throws appropriate error based on status code and configuration
+ * @returns {boolean} Returns true if the error was a 404 that should be ignored
+ */
+function _handleFileNotFoundError(error, bucket, key, crossCollectionThrowOnNotFound) {
+  if (!error || !isObject(error) || !('statusCode' in error)) {
+    throw error;
+  }
+  if (error.statusCode === HTTP_NOT_FOUND) {
+    if (crossCollectionThrowOnNotFound) {
+      throw new FileNotFound(
+        `File ${key} in bucket ${bucket} does not exist in the Cumulus database`
+      );
+    }
+
+    log.debug(
+      `File ${key} in bucket ${bucket} does not exist in the Cumulus database but exists in the filesystem. `
+      + 'Ignoring orphan due to MoveGranules configuration.'
+    );
+    return true;
+  }
+  throw error;
+}
+
+/**
+ * Validates that a cross-collection collision check can be performed.
+ *
+ * @param {string} granuleCollectionId - The collection ID to validate
+ * @param {string} bucket - S3 bucket name
+ * @param {string} key - S3 key
+ * @throws {ValidationError} If collection ID is missing
+ */
+function _validateCollisionCheckPreconditions(granuleCollectionId, bucket, key) {
+  if (!granuleCollectionId) {
```

Comment:
Is granuleCollectionId the collection id of the incoming granule? How could it be undefined?